src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -2592,6 +2592,16 @@ void InstanceKlass::remove_unshareable_info() {
   init_shared_package_entry();
   _dep_context_last_cleaned = 0;
   _init_monitor = nullptr;
+
+  remove_unshareable_flags();
+}
+
+void InstanceKlass::remove_unshareable_flags() {
+  // clear all the flags/stats that shouldn't be in the archived version
+#if INCLUDE_JVMTI
+  set_is_being_redefined(false);
```

Comment:
I think this should assert !is_scratch_class() and clear (?) is_redefined() also.  It's unfortunate that we can't just clear all the status flags here and in Method.

Reply:
> clear (?) is_redefined() also.

There is no such flag; did you mean `has_been_redefined`? Instead of clearing it, shouldn't it be an assert `!has_been_redefined()` as well?